### ISSUE TITLE
SVG Image used as Logo is duplicated in PDF for every page

### ISSUE THREAD

**Issue by jsf84ksnf** [Kozea/WeasyPrint#1978]
Cou can use an svg image as a logo ( @page { @top-right{ background-image: url("/static/logo.svg")}} ) and it works just fine.

But if the document has > 1 pages, the full SVG resource is included for every page in the pdfs, blowing up its size greatly. (100 pages includes the svg 100 times).

I also tried defining the logo as a svg-symbol with an id and then loading the id with svg-use, but that doesnt show up at all.

WeasyPrint Version: 60.1

**Comment by liZe** [comment]
> Is there a planned update for WeasyPrint to address this issue?

There’s nothing planned for this issue.

If you (or anyone) is interested in solving this issue, the first step is to find in the PDF specification a way to store the drawing once and reference it multiple times.

**Comment by ranma42** [comment]
@liZe IIUC an XObject could be used for SVG as well. From section 8.10.1 of PDF 32000-1:2008

>  A form XObject is a PDF content stream that is a self-contained description of any sequence of graphics 
> objects (including path objects, text objects, and sampled images). A form XObject may be painted multiple 
> times—either on several pages or at several locations on the same page—and produces the same results each 
> time, subject only to the graphics state at the time it is invoked. Not only is this shared definition economical to 
> represent in the PDF file, but under suitable circumstances the conforming reader can optimize execution by 
> caching the results of rendering the form XObject for repeated reuse.

It looks like in addition to raster images, form XObject could contain any kind of (composition of) paths, hence I believe they might be used for SVG images.

**Comment by liZe** [comment]
> It looks like form XObjects could contain any kind of (composition of) paths, hence I believe they might be used for SVG images.

Yes, it looks like a good idea. As raster images already use XObjects, we could use equivalent code for SVG and possibly merge some code between `RasterImage` and `SVGImage`.
